controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -219,6 +171,16 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 			r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
 		}
 
+		cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
+		cephBlockPoolListOption := []client.ListOption{
+			client.InNamespace(r.namespace),
+			client.MatchingLabels(map[string]string{StorageConsumerNameLabel: r.storageConsumer.Name}),
+		}
+
+		if err := r.List(r.ctx, cephBlockPoolList, cephBlockPoolListOption...); err != nil && len(cephBlockPoolList.Items) > 0 {
```

Comment:
+ `r.List` should not accept a context as the context is already on the reconciler. 
+ I don't see a list util on the reconciler, please add it
+ why does the util need to be public? we can just do `r.list` instead